pkg/build/strategies/layered/layered.go

```diff
@@ -146,6 +145,10 @@ func (builder *Layered) Build(config *api.Config) (*api.Result, error) {
 		return buildResult, fmt.Errorf("builder image uses ONBUILD instructions but ONBUILD is not allowed")
 	}
 
+	if config.BuilderImage == "" {
+		return nil, fmt.Errorf("builder image name cannot be empty")
```

Comment:
@jim-minter sorry to drop in on your PR, but can you return buildResult here with a generic s2i failure reason instead of nil?
i need openshift to read what kind of error happened in s2i and print it in the output together with the detailed message. something along the lines of:
```go
buildResult.BuildInfo.FailureReason = utilstatus.NewFailureReason(utilstatus.ReasonGenericS2IBuildFailed, utilstatus.ReasonMessageGenericS2iBuildFailed)
return buildResult, fmt.Errorf("builder image name cannot be empty")
```